### DISCUSSION TITLE
Developing Tic-tac-toe game

### DISCUSSION THREAD

**Discussion by ebajcar** [ebajcar/web10199_worksheets#13] [Weekly activities]
This week, we are working on developing a solution for a game of Tic Toc Toe.  We will incorporate the solution into our Course Website (see Preparing for Assignment 2).  The first phase is to analyze and design a solution.  Read, expand, and discuss:

[Developing tic-tac-toe app](https://www.notion.so/Developing-tic-tac-toe-app-5fa13367710f43cc81e98cd66b8a7369)

### It contains an obtrusive JS version. Copy, implement and document. 
Discuss here: What is good and bad about this solution
1. Is the solution using unobtrusive Javascript?
2. Is the solution using procedural, functional, or object-oriented paradigm?
3. Is the algorithm similar to what you have designed (chosen)? in what ways? 
4. Do you think all browsers will work the same?
5. Is the code understandable?  Describe in your own words.
6. Is the processing of Javascript delayed to accelerated to the event DOMContentLoaded?
7. Does the code use control boxes and event attributes?
8. How would you fix and improve the code to adhere to the principles of unobtrusive Javascript?

**Comment by DieuHuynhVn** [comment]
There are 2 players here. The game is played on a grid (square space e.g 3 squaresx3 squares, 5 squares x5 squares), developers can design more space.
My analysis for Tic Tac Toe
1/ The first player will start with the X sign and then the second player will start with the O sign. Players take turns putting their marks in empty squares.
2/Who gets 3 of marks in a row (up, down, across, or diagonally) is the winner. If they want to play the new game, click the "Reset Game" button to start new ones.
3/ 9 squares are full, the game is over. If no player has 3 marks, the game ends with no winner.

I downloaded the Tic Tac Toe version of the ma'am on ma'am GitHub and watched it.  From my point of view, it is a basic version and clears. We need to decorate the background, more colors and so on. It will be attractive. I will add a button " who will go first ?" when users click there, it will show randomly who will go first (X or O). That will be fair. Moreover, we should design more space ( e.g.10 squares 10 squares ) and the first player gets 3 in a row (up, down, across, or diagonally), 3 times is the winner. It is harder and makes users feel less boring by the short process ( 3 marks in a row (up, down, across) is the winner ). 

Above is my individual thinking to improve the code that ma'am provided on your GitHub.
